### DISCUSSION TITLE
ZepetoInputController 질문드립니다.

### DISCUSSION THREAD

**Discussion by Zimins** [naverz/zepeto-studio-kor#223] [5. Q&A]
안녕하세요~ zepeto script를 익혀보는 중입니다. 
ZepetoInputControl 에 이미 필요한 액션들이 정의되어 있어서, 해당 클래스를 이용하여 특정  GameObject의 움직임을 구현하려고 합니다. 

코드안에서 초기화 하는 코드를 여러 유니티 문서를 보고 진행해봤습니다. 

```
export default class DemoScript extends ZepetoScriptBehaviour {

    zepetoInputControl: ZepetoInputControl;

    Awake() {
        this.zepetoInputControl = new ZepetoInputControl();
    }

    OnEnable() {
        const callbacks: IPlayerActions = {
            OnMove: function ($context: CallbackContext): void {
                console.log("move");
            },
            OnJump: function ($context: CallbackContext): void {
            }
        };
        this.zepetoInputControl.Enable();
        this.zepetoInputControl.Player.SetCallbacks(callbacks);
    }

    OnDisable() {
        this.zepetoInputControl.Disable();
    }
}
```

이때 프로젝트를 실행하면
 System.Exception: file:///C:/Users/wlals/HideAndSeek/Assets/DemoScript.js:25: Error: invalid arguments to SetCallbacks
메시지가 표시되는데요, api 사용이 잘못된건가요? 혹은 다른 방법으로 사용자 입력을 받아야 하나요?

https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/manual/ActionAssets.html#auto-generating-script-code-for-actions
참고했던 링크입니다.

**Comment by moondory77-naverz** [comment]
안녕하세요.
ZepetoInputControl의 경우, Unity (New) InputSystem 모듈을 통해 정의된 InputActionAsset입니다.

다만, 현재 패키지 기준 (ZEPETO.World 1.1.2)
제페토스크립트에서, Unity (New) InputSystem 모듈에 대한 type declaration을 포함하고 있지 않기 때문에 
typescript 레벨에서 ZepetoInputControl에 접근하여, 임의로 커스터마이징하는 것은 불가한 상태입니다.

다음 업데이트를 통해 Unity (New) InputSystem 모듈에 대한 type declaration을 제공드릴 수 있도록 하겠습니다.
감사합니다.

**Reply by Zimins** [reply]
확인 감사합니다!
